pyaedt/modeler/cad/Primitives.py

```diff
@@ -8215,3 +8242,553 @@ def convert_segments_to_line(self, object_name):
+                elif csv_keys[col_cont] == "xc":
+                    props_cyl["Instances"][row_cont]["Origin"][0] = float(value)
+                elif csv_keys[col_cont] == "yc":
+                    props_cyl["Instances"][row_cont]["Origin"][1] = float(value)
+                elif csv_keys[col_cont] == "zc":
+                    props_cyl["Instances"][row_cont]["Origin"][2] = float(value)
+                elif csv_keys[col_cont] == "xd":
+                    props_cyl["Primitives"][row_cont]["X Length"] = float(value)
+                elif csv_keys[col_cont] == "yd":
+                    props_cyl["Primitives"][row_cont]["Y Length"] = float(value)
+                elif csv_keys[col_cont] == "zd":
+                    props_cyl["Primitives"][row_cont]["Z Length"] = float(value)
+                col_cont += 1
+            row_cont += 1
+        return props_box
+
+    @pyaedt_function_handler()
+    def convert_units(self, values):
+        """Convert input values to default units.
+
+        If the value has units, it is converted to a numeric value with the default units.
+
+        Parameters
+        ----------
+        values : list
+            List of values.
+
+        Returns
+        -------
+        list
+            List of numeric values.
+        """
+        extracted_values = []
+        for value in values:
+            if isinstance(value, (int, float)):
+                extracted_values.append(value)
+            elif isinstance(value, str):
+                value_number, units = decompose_variable_value(value)
+                if units:
+                    value_number = self._length_unit_conversion(value_number, units)
+                extracted_values.append(value_number)
+
+        return extracted_values
+
+    @pyaedt_function_handler()
+    def _length_unit_conversion(self, value, input_units):
+        """Convert value to input units."""
+        from pyaedt.generic.constants import unit_converter
+
+        converted_value = unit_converter(value, unit_system="Length", input_units=input_units, output_units=self.units)
+        return converted_value
+
+    @pyaedt_function_handler()
+    def _create_coordinate_system(self):
+        """Create coordinate system defined in the object."""
+        for cs in self.coordinate_systems:
+            cs_names = [cs.name for cs in self._app.modeler.coordinate_systems]
+            name = cs.get("Name")
+            if not name:
+                self.logger.warning("Coordinate system does not have Name parameter.")
+                return False
+            if name in cs_names:
+                self.logger.warning("Coordinate system {} already exists.".format(name))
+                continue
+            mode = cs.get("Mode")
+            if not mode or not any(key in mode for key in ["Axis/Position", "Euler Angle ZYZ", "Euler Angle ZXZ"]):
+                self.logger.warning(
+                    "Coordinate system does not have Mode parameter or it is not valid, "
```

Comment:
```suggestion
                    "Coordinate system does not have a 'Mode' parameter or it is not valid. "
```
Should Mode be capitalized